python/ray/tune/tune.py

```diff
@@ -340,6 +340,7 @@ def run(
     # a context object or similar.
     _tuner_api: bool = False,
     _trainer_api: bool = False,
+    _run_experiments_api: bool = False,
```

Comment:
This becomes a bit much with the flags. It also looks like we can consolidate them as only certain combinations are possible:

```
Trainer.fit() --> _tuner_api=True, _trainer_api=True
Tuner.fit() --> _tuner_api=True, _trainer_api=False
tune.run() --> _tuner_api=False, _trainer_api=False
```

Can we just consolidate these into one (still internal) attribute `_api_entrypoint` with values `[Trainer.fit, Tuner.fit, tune.run, tune.run_experiments]`?

Reply:
Done! Made it an enum. I kept the TuneController stuff as `_trainer_api`, since all things there are "for train".